integrationtests/self/key_update_test.go

```diff
@@ -7,14 +7,18 @@ import (
 	"time"
 
 	"github.com/quic-go/quic-go"
+	"github.com/quic-go/quic-go/internal/handshake"
 	"github.com/quic-go/quic-go/internal/protocol"
 	"github.com/quic-go/quic-go/logging"
 
 	"github.com/stretchr/testify/require"
 )
 
 func TestKeyUpdates(t *testing.T) {
-	t.Setenv("QUIC_GO_TEST_KEY_UPDATE_INTERVAL", "1") // update keys as frequently as possible
+	// update keys as frequently as possible
+	origKeyUpdateInterval := handshake.KeyUpdateInterval
+	handshake.KeyUpdateInterval = 1
+	t.Cleanup(func() { handshake.KeyUpdateInterval = origKeyUpdateInterval })
```

Comment:
And now you just introduced a race condition, see #5121. What are you even trying to achieve?!

Reply:
Ok, I think I don't understand the codebase properly.

If that test doesn't run in parallel, then it shouldn't race with changing the value. Or are there leftover goroutines that aren't waited at the end of tests that race with the values?

Reply:
Hmm, wouldn't that imply that there's still a logical race between setting the value via environment variables? But, I guess it doesn't really break anything to have it.